### PR TITLE
dasharo-compatibility/wifi-bluetooth-support.robot: atheros note

### PR DESCRIPTION
**Pull request by filipleple**

### CODE REVIEW

File: dasharo-compatibility/wifi-bluetooth-support.robot [hunk 1]
```diff
@@ -35,7 +39,7 @@ WLE001.001 Wireless card detection (Ubuntu 20.04)
     Login To Linux
     Switch To Root User
     ${out}=    Execute Command In Terminal    lspci | grep "Network controller:"
-    Should Match    ${out}    *${WIFI_CARD_UBUNTU}*
```
Comment (macpijan) [inline dasharo-compatibility/wifi-bluetooth-support.robot:38]: @filipleple I really do not like this change. You remove the `WIFI_CARD_UBUNTU`, keep the "dead" (not used) variable in configs, possibly also breaking this test for other platforms.

Comment (filipleple) [reply dasharo-compatibility/wifi-bluetooth-support.robot:38]: @macpijan I've reworked the test to be backwards compatible and run twice for each card

File: dasharo-compatibility/wifi-bluetooth-support.robot [hunk 2]
```diff
@@ -77,7 +77,70 @@ WLE002.002 Wi-Fi scanning (Windows 11)
...
+    Switch To Root User
+    Scan For Bluetooth In Linux
+    Exit From Root User
+
+WLE004.001 Wireless card detection (Ubuntu 20.04) (Atheros)
+    [Documentation]    Check whether the Wi-Fi/Bluetooth card is enumerated
+    ...    correctly and can be detected from the operating system.
+    Skip If    not ${WIRELESS_CARD_SUPPORT}    WLE001.001 not supported
+    Skip If    not ${TESTS_IN_UBUNTU_SUPPORT}    WLE001.001 not supported
+    Pause Execution In Console
```
Comment (filipleple) [inline dasharo-compatibility/wifi-bluetooth-support.robot:98]: keyword to be merged from Psotas, might highlight as not defined ATM

File: dasharo-compatibility/wifi-bluetooth-support.robot [hunk 3]
```diff
@@ -35,20 +39,20 @@ WLE001.001 Wireless card detection (Ubuntu 20.04)
     Login To Linux
     Switch To Root User
     ${out}=    Execute Command In Terminal    lspci | grep "Network controller:"
-    Should Match    ${out}    *${WIFI_CARD_UBUNTU}*
+    Should Contain    ${out}    *${WIFI_CARD_UBUNTU}*
     Exit From Root User
 
-WLE001.002 Wireless card detection (Windows 11)
+WLE001.002 Wireless card detection (Windows 11) (Intel)
```
Comment (Psotas) [inline dasharo-compatibility/wifi-bluetooth-support.robot:45]: You shouldn't change it.
For protectli-v1*10 the variable values are:
```
${WIFI_CARD}=                                       Qualcomm
${WIFI_CARD_UBUNTU}=                                Qualcomm
```

File: dasharo-compatibility/wifi-bluetooth-support.robot [hunk 4]
```diff
@@ -73,7 +77,70 @@ WLE002.002 Wi-Fi scanning (Windows 11)
...
+    Skip If    not ${WIRELESS_CARD_BLUETOOTH_SUPPORT}    WLE003.001 not supported
+    Skip If    not ${TESTS_IN_UBUNTU_SUPPORT}    WLE003.001 not supported
+    Power On
+    Boot System Or From Connected Disk    ubuntu
+    Login To Linux
+    Switch To Root User
+    Scan For Bluetooth In Linux
+    Exit From Root User
+
+WLE004.001 Wireless card detection (Ubuntu 20.04) (Atheros)
```
Comment (Psotas) [inline dasharo-compatibility/wifi-bluetooth-support.robot:93]: This is still a variant of the test:
`WLE001.00X Wireless card detection`
So the test number should remain the same, only the variant number should change.
```suggestion
WLE001.003 Wireless card detection (Ubuntu 20.04) (Atheros)
```

File: dasharo-compatibility/wifi-bluetooth-support.robot (hunk 1 above)
Comment (Psotas) [reply dasharo-compatibility/wifi-bluetooth-support.robot:38]: > I think we can also have a single entry in the results sheet, if that helps. The PASS would mean both passed. The FAIL would mean that any one of these failed, and would need to be clarified as a comment.

@macpijan cc
@filipleple
This should be the simplest solution.

So leave the original tests.
Add reminder information:
```
Log To Console    Remember to test all variants of wireless cards.
Log    Remember to test all variants of wireless cards.    WARN
```
Add to the information which card was detected.
```
Log To Console    The test passed for the XXX wireless card
Log    The test passed for the XXX wireless card    WARN
```

Of course, these are just suggestions. If you have better ideas, go ahead.

**Comment by filipleple** [comment]
Reworked in accordance with @Psotas 's suggestion - one set of test cases, each warns the tester to test all cards and logs what is the current card.

File: dasharo-compatibility/wifi-bluetooth-support.robot [hunk 5]
```diff
@@ -30,61 +34,83 @@ WLE001.001 Wireless card detection (Ubuntu 20.04)
...
     ...    correctly and can be detected from the operating system.
     Skip If    not ${WIRELESS_CARD_SUPPORT}    WLE001.002 not supported
     Skip If    not ${TESTS_IN_WINDOWS_SUPPORT}    WLE001.002 not supported
+    Log To Console    Remember to test all variants of wireless cards.
+    Log    Remember to test all variants of wireless cards.    WARN
     Power On
     Login To Windows
     ${out}=    Execute Command In Terminal    Get-PnpDevice -PresentOnly | Select-String -Pattern "Wi-Fi"
-    Should Contain    ${out}    ${WIFI_CARD}
+    Should Match    ${out}    *${WIFI_CARD_UBUNTU}*
```
Comment (Psotas) [inline dasharo-compatibility/wifi-bluetooth-support.robot:59]: Shouldn't it be ${WIFI_CARD}?

File: dasharo-compatibility/wifi-bluetooth-support.robot [hunk 6]
```diff
@@ -30,61 +34,83 @@ WLE001.001 Wireless card detection (Ubuntu 20.04)
...
+    Log To Console    Remember to test all variants of wireless cards.
+    Log    Remember to test all variants of wireless cards.    WARN
     Power On
     Login To Windows
     Execute Command In Terminal    Start-Service WlanSvc
     ${out}=    Execute Command In Terminal    netsh wlan show network
     Should Contain    ${out}    3mdeb_abr
     Should Contain    ${out}    3mdeb_abr_5GHz
+    Log To Console    The test passed for the ${WIFI_CARD} wireless card
+    Log    The test passed for the ${WIFI_CARD} wireless card    WARN
```
Comment (Psotas) [inline dasharo-compatibility/wifi-bluetooth-support.robot:95]: I don't know if it works, but apparently you can check the name of the wifi card.
https://learn.microsoft.com/en-us/powershell/module/netadapter/get-netadapter?view=windowsserver2022-ps

File: dasharo-compatibility/wifi-bluetooth-support.robot (hunk 2 above)
Comment (filipleple) [reply dasharo-compatibility/wifi-bluetooth-support.robot]: -- outdated, doesn't contain the keyword anymore

File: dasharo-compatibility/wifi-bluetooth-support.robot (hunk 5 above)
Comment (filipleple) [reply dasharo-compatibility/wifi-bluetooth-support.robot]: Fixed [here](https://github.com/Dasharo/open-source-firmware-validation/pull/136/files#diff-db99b3a0adebbc880d28cd1f87d49ea87c9b11cd1d3681f599bd1494c26e814cR59)

File: dasharo-compatibility/wifi-bluetooth-support.robot (hunk 6 above)
Comment (filipleple) [reply dasharo-compatibility/wifi-bluetooth-support.robot]: [Implemented](https://github.com/Dasharo/open-source-firmware-validation/pull/136/files#diff-db99b3a0adebbc880d28cd1f87d49ea87c9b11cd1d3681f599bd1494c26e814cR61) for Windows test cases

File: dasharo-compatibility/wifi-bluetooth-support.robot (hunk 4 above)
Comment (filipleple) [reply dasharo-compatibility/wifi-bluetooth-support.robot]: Test case removed

File: dasharo-compatibility/wifi-bluetooth-support.robot (hunk 3 above)
Comment (filipleple) [reply dasharo-compatibility/wifi-bluetooth-support.robot]: [Names](https://github.com/Dasharo/open-source-firmware-validation/pull/136/files#diff-db99b3a0adebbc880d28cd1f87d49ea87c9b11cd1d3681f599bd1494c26e814cL41) no longer changed

File: dasharo-compatibility/wifi-bluetooth-support.robot (hunk 1 above)
Comment (Psotas) [reply dasharo-compatibility/wifi-bluetooth-support.robot:38]: Looks good to me.

**Review by Psotas — APPROVED** [review]

**Pull request merged**